pkg/registry/apis/provisioning/jobs/syncer.go

```diff
@@ -201,7 +154,10 @@ func sortResourceListForDeletion(list *provisioning.ResourceList) {
 	})
 }
 
-func (r *Syncer) deleteAllProvisionedResources(ctx context.Context) error {
+const CLEANUP_FINALIZER = "cleanup"
+
+// Called by the cleanup finalizer
+func (r *Syncer) DeleteAllProvisionedResources(ctx context.Context) error {
```

Comment:
Just a side note: WDYT about setting `ownerReferences` and using them alongside finalizers to cascade-delete all resources owned by the repository? It feels more standard/K8s-like pattern to me.